### ISSUE TITLE
[CI] Actions should only pull go deps once

### ISSUE THREAD

**Issue by gregdhill** [hyperledger/burrow#1282]
In a couple of the [test steps](https://github.com/hyperledger/burrow/blob/master/.github/workflows/test.yaml) that require docker builds we pull in our go dependencies twice. We should ideally vendor these first and reuse to speed up the pipeline.

**Comment by deepakchethan** [comment]
According to actions/setup-go. It automatically caches go, which can be seen [here](https://github.com/actions/setup-go/blob/master/src/installer.ts). So doesn't vendoring this first make it redundant?

**Comment by gregdhill** [comment]
Correct, however this unfortunately is not the case with our docker build steps which do not share this cache.
